packages/analyticsdx-template-lint/src/validator.ts

```diff
@@ -0,0 +1,237 @@
+/*
+ * Copyright (c) 2021, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import * as fs from 'fs';
+import { JSONPath, Node as JsonNode } from 'jsonc-parser';
+import * as path from 'path';
+import {
+  Diagnostic,
+  DiagnosticSeverity,
+  getLanguageService,
+  JSONSchema,
+  LanguageService,
+  Range,
+  TextDocument
+} from 'vscode-json-languageservice';
+import { DiagnosticRelatedInformation, Location } from 'vscode-languageserver-types';
+import { LINTER_SOURCE_ID } from './constants';
+import { TemplateLinter, TemplateLinterDiagnosticSeverity } from './linter';
+import { schemas } from './schemas';
+import { matchJsonNodesAtPattern } from './utils';
+
```

Comment:
This is the other interesting code -- it uses the same `vscode-json-languageservice` that the vscode json language server users for json schema validation.